pandas/io/tests/test_excel.py

```diff
@@ -92,110 +95,122 @@ def read_csv(self, *args, **kwds):
+                        parse_cols=[0, 2, 3])
+        # TODO add index to xls file)
+        tm.assert_frame_equal(df, dfref, check_names=False)
+        tm.assert_frame_equal(df3, dfref, check_names=False)
 
     def test_parse_cols_str(self):
-        _skip_if_no_openpyxl()
-        _skip_if_no_xlrd()
 
-        suffix = ['xls', 'xlsx', 'xlsm']
-
-        for s in suffix:
-
-            pth = os.path.join(self.dirpath, 'test.%s' % s)
-            xls = ExcelFile(pth)
-
-            df = xls.parse('Sheet1', index_col=0, parse_dates=True,
-                           parse_cols='A:D')
-            df2 = read_csv(self.csv1, index_col=0, parse_dates=True)
-            df2 = df2.reindex(columns=['A', 'B', 'C'])
-            df3 = xls.parse('Sheet2', skiprows=[1], index_col=0,
-                            parse_dates=True, parse_cols='A:D')
-            # TODO add index to xls, read xls ignores index name ?
-            tm.assert_frame_equal(df, df2, check_names=False)
-            tm.assert_frame_equal(df3, df2, check_names=False)
-            del df, df2, df3
-
-            df = xls.parse('Sheet1', index_col=0, parse_dates=True,
-                           parse_cols='A,C,D')
-            df2 = read_csv(self.csv1, index_col=0, parse_dates=True)
-            df2 = df2.reindex(columns=['B', 'C'])
-            df3 = xls.parse('Sheet2', skiprows=[1], index_col=0,
-                            parse_dates=True,
-                            parse_cols='A,C,D')
-            # TODO add index to xls file
-            tm.assert_frame_equal(df, df2, check_names=False)
-            tm.assert_frame_equal(df3, df2, check_names=False)
-            del df, df2, df3
-
-            df = xls.parse('Sheet1', index_col=0, parse_dates=True,
-                           parse_cols='A,C:D')
-            df2 = read_csv(self.csv1, index_col=0, parse_dates=True)
-            df2 = df2.reindex(columns=['B', 'C'])
-            df3 = xls.parse('Sheet2', skiprows=[1], index_col=0,
-                            parse_dates=True,
-                            parse_cols='A,C:D')
-            tm.assert_frame_equal(df, df2, check_names=False)
-            tm.assert_frame_equal(df3, df2, check_names=False)
+        pth = os.path.join(self.dirpath, 'test1' + self.ext)
+        pref = os.path.join(self.dirpath, 'test1.csv')
+        # the truth is obtained form read_csv with Python engine
+        dfref = self.read_csv(pref, index_col=0, parse_dates=True)
+
+        xls = ExcelFile(pth)
+        df = xls.parse('Sheet1', index_col=0, parse_dates=True,
+                       parse_cols='A:D')
+        df2 = dfref.reindex(columns=['A', 'B', 'C'])
+        df3 = xls.parse('Sheet2', skiprows=[1], index_col=0,
+                        parse_dates=True, parse_cols='A:D')
+        # TODO add index to xls, read xls ignores index name ?
+        tm.assert_frame_equal(df, df2, check_names=False)
+        tm.assert_frame_equal(df3, df2, check_names=False)
+        del df, df2, df3
```

Comment:
why are you explicitly deleted these variables? generally there's no reason to do that in python....